File: Loenn/entities/dash_code_controller.lua
local dashCodeController = {}

dashCodeController.name = "vitellary/dashcodecontroller"
dashCodeController.depth = -100

dashCodeController.fieldInformation = {
    index = {
        fieldType = "integer",
    },
}

dashCodeController.placements = {
    {
        name = "dash_code_controller",
        data = {
            dashCode = "",
            flagLabel = "",
            flagOnFailure = "",
            index = 0,
        }
    }
}

dashCodeController.texture = "ahorn_dashcodecontroller"

return dashCodeController

File: Ahorn/entities/dashCodeController.jl
module VitellaryDashSequenceController

using ..Ahorn, Maple

@mapdef Entity "vitellary/dashcodecontroller" DashController(x::Integer, y::Integer,
    dashCode::String="", flagLabel::String="", flagOnFailure::String="", index::Integer=0)

const placements = Ahorn.PlacementDict(
    "Dash Code Flag Sequence Controller (Crystalline)" => Ahorn.EntityPlacement(
        DashController
    )
)

function Ahorn.selection(entity::DashController)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle("ahorn_dashcodecontroller", x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DashController)
    Ahorn.drawSprite(ctx, "ahorn_dashcodecontroller", 0, 0)
end

end